src/classes/COFF_PreventUpdate_TDTM.cls

```diff
@@ -0,0 +1,72 @@
+/*
+    Copyright (c) 2019, Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group Course Offerings
+* @group-content ../../ApexDocContent/CourseOfferings.htm
+* @description This class prevents changes to the Course lookup field on 
+* Course Offering if there are Term Grades associated with it.
+*/
+public with sharing class COFF_PreventUpdate_TDTM extends TDTM_Runnable {
+
+    /*******************************************************************************************************
+    * @description This class prevents changes to the Course lookup field on 
```

Comment:
I'm reviewing the spec and started to review this code but it's not clear to me why we care about preventing the Course lookup field from changing. The Course isn't stored on the Term Grade so it shouldn't really matter. Let's talk about this with @lifewithryan sometime tomorrow or next week.

Reply:
Forgot about this - can we set up a quick meeting to discuss this with @lifewithryan? (or maybe he'll see this and chime in here :))

Reply:
@skosgi1 I discussed this with @lifewithryan this morning and we came to the agreement that we should remove this functionality from the PR for now (both this and the unit test class).

I'll update the WI and tech specs momentarily. Sorry for the unnecessary work! We discussed that the better solution might be to prevent the Course from being changed on the Course Offering if there are Course Connections -- so it's unrelated to the Term Grades work and I think we need more discussion on it before we proceed. This will likely have its own tech spec and work item.

Reply:
It's not that it's unnecessary really, I think ultimately we will have to do something in the future. Maybe keep this around as a file on your local machine a bit -- it may come back on some shape or form.

Reply:
@lifewithryan Good point! It might even be worth spinning these few files into their own branch for a bit.